File: python/robot-name/robot_name.py
from random import randint

class Robot(object):
    name_list = []
    def __init__(self):
        self.reset()
    
    def reset(self):
        name = ''
        for i in range(5):
            if (i < 2):
                name += chr(randint(ord('A'), ord('Z')))
            else:
                name += str(randint(0, 9))

        if (name in self.name_list):
            self.reset()
        else:
            self.name = name
            self.name_list.append(name)

File: cpp/robot-name/robot_name.cpp
#include "robot_name.h"

namespace robot_name {
  robot::robot() {
    reset();
  }

  string robot::name() const {
    return robot_name;
  }

  void robot::reset() {
    string new_name = "";
    for (int i = 1; i <= 5; i++) {
      if (i < 3) {
        // append a random uppercase letter
        new_name += char(rand() % 26 + 65);
      } else {
        // append a random number
        new_name += to_string(rand() % 9);
      }
    }
    // check to see if this name has already been stored
    if (find(name_list.begin(), name_list.end(), new_name) != name_list.end()) {
      reset();
    } else {
      robot_name = new_name;
      name_list.push_back(new_name);
    }
  }
}
